miio/airhumidifier.py

```diff
@@ -198,11 +198,11 @@ def depth(self) -> Optional[int]:
     def water_level(self) -> Optional[int]:
         """Return current water level in percent.
 
-        If water tank is full, depth is 125.
+        If water tank is full, depth is 120. If water tank is overfilled, depth is 125.
         """
         depth = self.data.get("depth")
         if depth is not None and depth <= 125:
-            return int(depth / 1.25)
+            return int(max(min(depth / 1.2, 100), 0))
```

Comment:
Why use `max(val, 0)` here, avoiding negative values?

Reply:
Because we have this
```
        else if (value === -1) {
          value = 0;
        }
```
in tha app source.

Reply:
Sorry for the delay, I see. I think it would be the best to inverse the condition and bail out early
```
if depth is None or depth < 0:
    return None

return <real value here>
```

Reply:
I did simplify the logic a bit. If it looks okay to you, I think this is ready to be merged after the tests are passing.